### ISSUE TITLE
Better handling of tags inside dynamic includes

### ISSUE THREAD

**Issue by Vladimir-csp** [ansible/ansible#38262]
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

core
##### ANSIBLE VERSION

```
2.5
```

##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A
##### SUMMARY

With ansible 2.5 tags on dynamic includes no longer inherited by tasks inside dynamic includes.
I think there is a better non-disruptive solution:
1. Keep `tags:` inheritable, so it would not be a behavioral exception limited to dynamic includes. 
2. Add new parameter to attach tags non-inheritantly, like this:
```
- name: some dynamic stuff
  include_tasks: "base_config_{{ ansible_distribution }}.yaml"
  # these tags will be inherited by all underlying tasks
  tags:
    - base
  # these tags inform ansible-playbook that it may find some of them inside
  lookup_tags:
    - logind
    - systemd
    - rcd
```

**Comment by bcoca** [comment]
We are looking at a feature that would set keywords for the included tasks to inherit, not just for tags:

```
- include_task: 
     file: whtvr.yml
     inherit: #let the bikeshed begin
          tags: ['other']
          become: false
          delegate_to: localhost
          when: whytorunincludedtasks|bool
  tags: ['doinclude']
  when: whytoruntheinclude|bool
```

**Comment by skylerbunny** [comment]
> We are looking at a feature...

... that definitely would be better than how 2.5 has left things, which is 'you can't use dynamic includes with tags unless you tag all subordinate tasks they contain, either directly or with blocks'. From a practical standpoint, I'd be happy to see #35735 rendered obsolete by this behavior being changed as proposed above.

**Comment by Vladimir-csp** [comment]
`inherit:` looks good, but it is also too disruptive compared to 2.4 behavior. Maybe, use `not_inherit:` instead?

**Comment by skylerbunny** [comment]
For the new feature proposal, if include_[tasks|role] executed with -t on the command line, and `tags:` returns zero tasks, wouldn't it be sufficient/appropriate to return a warning?

[Warning]: Dynamic include used with `tags:` found zero tasks. Are any included tasks tagged?

This might actually be a good idea even if the new tag inheritance feature is never merged in, if it's possible to warn on this eventuality.

**Comment by skylerbunny** [comment]
This might even want to be considered a hard-stop error, if there aren't any realistic circumstances in which a user would (tag an include block) and also (tag nothing inside that include). Put another way, is this essentially a usage error at that point?

**Comment by sivel** [comment]
I have 2 different implementations of the feature (targeted for 2.7) to resolve this issue:

* https://github.com/ansible/ansible/pull/39236 (implementation most likely to be accepted)
* https://github.com/ansible/ansible/pull/39237

**Comment by sivel** [comment]
This functionality has been implemented in https://github.com/ansible/ansible/pull/39236 and will be included in the 2.7 release.
